src/blocks/scratch3_event.js

```diff
@@ -90,10 +90,12 @@ class Scratch3EventBlocks {
     }
 
     broadcastAndWait (args, util) {
-        const broadcastVar = util.runtime.getTargetForStage().lookupBroadcastMsg(
-            args.BROADCAST_OPTION.id, args.BROADCAST_OPTION.name);
-        if (broadcastVar) {
-            const broadcastOption = broadcastVar.name;
+        if (!util.stackFrame.broadcastVar) {
+            util.stackFrame.broadcastVar = util.runtime.getTargetForStage().lookupBroadcastMsg(
+                args.BROADCAST_OPTION.id, args.BROADCAST_OPTION.name);
+        }
+        if (util.stackFrame.broadcastVar) {
```

Comment:
Ah yes, you're right! Good catch. I guess it's written in a confusing way. It could be clearer to write the two checks as:

if (!util.stackFrame.broadcastVar) {
    // attempt to set it
}

// If we failed to find the broadcast message, exit
if (!util.stackFrame.broadcastVar) return;

But for the sake of simplicity of the code change from what we had before, this is fine too.